packages/api/src/hooks/useRemoteConfig.ts

```diff
@@ -15,12 +16,25 @@ const remoteConfigQuery = async function () {
 };
 
 function useRemoteConfig(enabled = false) {
-    return useQuery({
-        queryKey: ['remoteConfig'],
-        queryFn: remoteConfigQuery,
-        initialData: initData,
-        enabled,
-    });
+    const [data, setData] = useState(initData);
+
+    useEffect(() => {
+        enabled &&
+            remoteConfigQuery()
+                .then(async res => {
+                    const resHash = await ObjectUtils.hashObject(res);
+                    const dataHash = await ObjectUtils.hashObject(data);
+                    if (resHash !== dataHash) {
+                        setData(res);
+                    }
+                })
+                .catch(error => {
+                    // eslint-disable-next-line no-console
+                    console.log('Remote Config error: ', error);
+                });
+    }, [enabled]);
```

Comment:
```suggestion
    }, [enabled, data]);
```

@sergei-deriv `data` is missing from the dependency array and you're only relying on enabled, which will stay same and you'll not get the updated data from the state so the comparison can be false every subsequent checks.